Source/WebCore/platform/ScrollSnapAnimatorState.cpp

```diff
@@ -95,18 +95,72 @@ float ScrollSnapAnimatorState::adjustedScrollDestination(ScrollEventAxis axis, F
     return offset * pageScale;
 }
 
+void ScrollSnapAnimatorState::setFocusedElementForAxis(ScrollEventAxis axis)
+{
+    auto snapOffsets = snapOffsetsForAxis(axis);
+    auto found = std::find_if(snapOffsets.begin(), snapOffsets.end(), [](SnapOffset<LayoutUnit> p) -> bool {
+        return p.isFocused;
+    });
+    if (found != snapOffsets.end()) {
```

Comment:
I have a slight preference for an early return here.

Reply:
Fixed.